tests/v2/test_waku_discv5.nim

```diff
@@ -309,3 +309,77 @@ procSuite "Waku Discovery v5":
     assert emptyRes.isOk(), emptyRes.error
     assert emptyRes.value.isNone(), $emptyRes.value
 
+  asyncTest "filter peer per static shard":
+    ## Given
+    let recordCluster21 = block:
+      let
+        enrSeqNum = 1u64
+        enrPrivKey = generatesecp256k1key()
+
+      let
+        shardCluster: uint16 = 21
+        shardIndices: seq[uint16] = @[1u16, 2u16, 5u16, 7u16, 9u16, 11u16]
+
+      let shards = RelayShards.init(shardCluster, shardIndices)
+
+      var builder = EnrBuilder.init(enrPrivKey, seqNum = enrSeqNum)
+      require builder.withWakuRelaySharding(shards).isOk()
+
+      let recordRes = builder.build()
+      require recordRes.isOk()
+      recordRes.tryGet()
+
+    let recordCluster22Indices1 = block:
+      let
+        enrSeqNum = 1u64
+        enrPrivKey = generatesecp256k1key()
+
+      let
+        shardCluster: uint16 = 22
+        shardIndices: seq[uint16] = @[2u16, 4u16, 5u16, 8u16, 10u16, 12u16]
+
+      let shards = RelayShards.init(shardCluster, shardIndices)
+
+      var builder = EnrBuilder.init(enrPrivKey, seqNum = enrSeqNum)
+      require builder.withWakuRelaySharding(shards).isOk()
+
+      let recordRes = builder.build()
+      require recordRes.isOk()
+      recordRes.tryGet()
+
+    let recordCluster22Indices2 = block:
+      let
+        enrSeqNum = 1u64
+        enrPrivKey = generatesecp256k1key()
+
+      let
+        shardCluster: uint16 = 22
+        shardIndices: seq[uint16] = @[1u16, 3u16, 6u16, 7u16, 9u16, 11u16]
+
+      let shards = RelayShards.init(shardCluster, shardIndices)
+
+      var builder = EnrBuilder.init(enrPrivKey, seqNum = enrSeqNum)
+      require builder.withWakuRelaySharding(shards).isOk()
+
+      let recordRes = builder.build()
+      require recordRes.isOk()
+      recordRes.tryGet()
+
```

Comment:
`#When` ? (Trying to follow the GIVEN-WHEN-THEN pattern and missing the "When" :D )